Orange/preprocess/transformation.py

```diff
@@ -107,7 +107,10 @@ def __init__(self, variable, offset, factor):
         self.factor = factor
 
     def transform(self, c):
-        return (c - self.offset) * self.factor
+        if sp.issparse(c):
+            return c
```

Comment:
I don't like normalizing of non-zero values only, since you're putting them on a different scale than zero-values. What about just multiplying the (non-zero) values by `self.factor`?

This of course only makes sense when the factor is computed with offset=0. We can either accept that sometimes the transformation won't make sense or, perhaps better, raise an exception of `self.offset != 0`.

```
if sp.issparse(c):
    if self.offset != 0:
        raise ValueError("Non-zero offset in normalization of sparse data")
    return c * self.factor  # ... or whatever is the optimal way to multiply a sparse matrix with a scalar
```